src/event_logger.cpp

```diff
@@ -110,4 +110,76 @@ void EventLogger::createAsyncPelWithI2cBusCallout(
     (void)i_userData1;
     (void)i_userData2;
 }
+
+void EventLogger::createAsyncPel(
+    const types::ErrorType i_errorType, const types::SeverityType i_severity,
+    const std::string& i_fileName, const std::string& i_funcName,
+    const std::string& i_internalRc,
+    const std::optional<std::pair<std::string, std::string>> i_userData1,
+    const std::optional<std::pair<std::string, std::string>> i_userData2)
+{
+    try
+    {
+        if (m_errorMsgMap.find(i_errorType) == m_errorMsgMap.end())
+        {
+            logging::logMessage(
+                "PEL error type not found in the error message map.");
+            return;
+        }
+
+        const std::string& l_message = m_errorMsgMap.at(i_errorType);
+
+        const std::string& l_severity =
+            (m_severityMap.find(i_severity) != m_severityMap.end()
+                 ? m_severityMap.at(i_severity)
+                 : m_severityMap.at(types::SeverityType::Error));
+
+        std::string l_description("Description field missing in user data");
+
+        if (i_userData1 && ((*i_userData1).first.compare("DESCRIPTION") ==
+                            constants::STR_CMP_SUCCESS))
+        {
+            l_description = (*i_userData1).second;
+        }
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        auto l_rc = -1;
+        if (!i_userData2)
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 4, "FileName",
+                i_fileName.c_str(), "FunctionName", i_funcName.c_str(),
+                "InternalRc", i_internalRc.c_str(), "DESCRIPTION",
+                l_description.c_str());
+        }
+        else
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 5, "FileName",
```

Comment:
Can magic number 5 be described by a `constexpr`  or `macro`?

Reply:
kept it as const local variable.